### PR TITLE
Make new apps layer datasets loadable when installing SF

### PR DESCRIPTION
**Pull request by trbromley**
The apps-layer datasets are currently not available on fresh installs of SF that install from github or (in-future) PyPI. This is because we changed the location of the data files in https://github.com/XanaduAI/strawberryfields/pull/401 and forgot to update the locations in `MANIFEST.in` and `setup.py`. We didn't notice this and our tests passed because we were testing on installs of SF from source (i.e., the cloned repo, which includes the data files).

This PR fixes the problem. For example, let's install the SF master branch:
```bash
pip uninstall strawberryfields -y
pip install git+https://github.com/XanaduAI/strawberryfields.git
```
In Python, we get the error:
```python
>>> from strawberryfields.apps import data
>>> data.Planted()
FileNotFoundError: [Errno 2] No such file or directory
```

Now let's see if this PR fixes things:
```bash
pip uninstall strawberryfields -y
pip install git+https://github.com/XanaduAI/strawberryfields.git@fix_data_loading
```
Now, in Python things work:
```python
>>> from strawberryfields.apps import data
>>> data.Planted()
<strawberryfields.apps.data.sample.Planted object at 0x7f99a04afa90>
```
Please also try the above!

### CODE REVIEW

**Comment by antalszava** [comment]
Couldn't reproduce the `FileNotFoundError` :thinking: tried installing the master branch of `StrawberryFields` with `pip install -e .` too.

```
Strawberry Fields: a Python library for continuous-variable quantum circuits.
Copyright 2018-2020 Xanadu Quantum Technologies Inc.

Python version:            3.7.4
Platform info:             Linux-5.3.0-59-generic-x86_64-with-debian-buster-sid
Installation path:         /Documents/code_base/strawberryfields/strawberryfields
Strawberry Fields version: 0.15.0-dev0
Numpy version:             1.19.0
Scipy version:             1.4.1
SymPy version:             1.6
NetworkX version:          2.3
The Walrus version:        0.13.0rc1
Blackbird version:         0.2.3
TensorFlow version:        2.0.1
```

**Comment by trbromley** [comment]
Thanks @antalszava! I should add - this happens when you are not in the main working directory of the SF repo - in that case it can still see the files.

**Comment by josh146** [comment]
This is a strong argument to moving to tox. Since tox installs the package in a fresh virtual environment, it acts as a test against our packaging.

File: setup.py [hunk 1]
```diff
@@ -44,7 +44,8 @@
     "url": "https://github.com/XanaduAI/StrawberryFields",
     "license": "Apache License 2.0",
     "packages": find_packages(where="."),
-    "package_data": {"strawberryfields": ["backends/data/*", "apps/data/*"]},
+    "package_data": {"strawberryfields": ["backends/data/*", "apps/data/feature_data/*",
+                                          "apps/data/sample_data"]},
```
Comment (josh146) [inline setup.py:48]: How come `apps/data/feature_data/*` with asterisk is listed, but `apps/data/sample_data` does not include an asterisk?

Comment (trbromley) [inline setup.py:48]:
```suggestion
                                          "apps/data/sample_data/*"]},
```
Good spot!

Comment (trbromley) [reply setup.py:48]:
```suggestion
                                          "apps/data/sample_data/*"]},
```
Good spot!

**Comment by AroosaIjaz** [comment]
Dear @trbromley, thank you for checking this. I do not get the file error even if I try from other directories. Do you think it is related to if we have the latest SF master version installed or not? I will wait for @ixfoduap 's feedback too.

**Comment by trbromley** [comment]
> Dear @trbromley, thank you for checking this. I do not get the file error even if I try from other directories. Do you think it is related to if we have the latest SF master version installed or not? I will wait for @ixfoduap 's feedback too.

Thanks @AroosaIjaz! Yes, this has to be from a fresh install of SF from the github master such as
```bash
pip uninstall strawberryfields -y
pip install git+https://github.com/XanaduAI/strawberryfields.git
```

**Comment by AroosaIjaz** [comment]
> > Dear @trbromley, thank you for checking this. I do not get the file error even if I try from other directories. Do you think it is related to if we have the latest SF master version installed or not? I will wait for @ixfoduap 's feedback too.
> 
> Thanks @AroosaIjaz! Yes, this has to be from a fresh install of SF from the github master such as
> 
> ```shell
> pip uninstall strawberryfields -y
> pip install git+https://github.com/XanaduAI/strawberryfields.git
> ```

Ah! Now it works as you suggest! Thanks!

**Pull request merged**